collect_app/src/main/java/org/odk/collect/android/audio/PlaybackFailedException.java

```diff
@@ -2,18 +2,32 @@
 
 import androidx.core.util.ObjectsCompat;
 
+import org.odk.collect.android.R;
+
+import java.io.File;
+
 public class PlaybackFailedException extends Exception {
 
     private final String uri;
+    private final int exceptionMessage;
 
     public PlaybackFailedException(String uri) {
         this.uri = uri;
+        this.exceptionMessage = setUpExceptionMessage();
     }
 
     public String getURI() {
         return uri;
     }
 
+    private int setUpExceptionMessage() {
+        return new File(uri).exists() ? R.string.file_invalid : R.string.file_missing;
```

Comment:
I think it would be nicer for this decision to be made inside `AudioPlayerViewModel` where the error occurs. That way we don't need to care about reading the file in `ODKView`. What do you think?